app/models/miq_queue.rb

```diff
@@ -246,6 +246,23 @@ def self.submit_job(options)
     put(options)
   end
 
+  # For services with a dedicated queue_name and worker, such as reporting, high priority work is
+  # run in the 'generic' queue with HIGH_PRIORITY so priority workers or even generic workers can
+  # pick it up immediately. Anything but high priority work or higher is run in the service's
+  # queue, such as 'reporting', and the dedicated worker will process this message following
+  # priority processing.
+  #
+  # Why? Reports are often started by a user who is actively waiting for the result so we need a way
+  # to expedite their request in workers who deal with high priority work.  At the same time, if a
+  # user isn't actively waiting for a report, it should be handled by reporting workers and not sit
+  # behind other items in the generic queue as it's far easier to scale these workers up and down
+  # as needed.
+  #
+  # TODO: Review if other services in submit_job, such as event/smart proxy should follow this pattern.
+  def self.queue_name_for_priority_service(service, priority)
+    (priority || NORMAL_PRIORITY) <= HIGH_PRIORITY ? "generic" : service.to_s
```

Comment:
This is hard to read to me. I'd say prefer the MiqQueue.priority set of methods so there's no guessing, but that method would only check `< HIGH_PRIORITY`.  Wonder if we need _another_ method there, if only for readability.

Not my suggestion, but using the method would look like the following if we need `<=`:

```suggestion
    high_priority_or_higher = MiqQueue.higher_priority?(priority || NORMAL_PRIORITY, MiqQueue.priority(HIGH_PRIORITY, :lower, 1))
    high_priority_or_higher ? "generic" : service.to_s
```

Alternatively, if we only need `<`, then

```suggestion
    higher_than_high_priority = MiqQueue.higher_priority?(priority || NORMAL_PRIORITY, HIGH_PRIORITY)
    higher_than_high_priority ? "generic" : service.to_s
```


I kind of wish I made this column it's own class that has methods like `.higher` / `.lower`, etc.

Reply:
yeah, I can do that.  I figured it was ok for the this class to know how the priority values work but agree `higher_priority?` is more readable.

Reply:
I have to do `lower_priority?` and swap the values since we want HIGH_PRIORITY or higher to get on the generic queue.  `high_priority?` is false when the two priorities are both high.

Reply:
That's perfect - thanks!